src/rust/inetstack/protocols/tcp/peer.rs

```diff
@@ -207,13 +207,15 @@ impl<const N: usize> SharedTcpPeer<N> {
     /// Sets up the coroutine for accepting a new connection.
     pub fn accept(&mut self, qd: QDesc) -> Result<QToken, Fail> {
         trace!("accept(): qd={:?}", qd);
+
         let mut queue: SharedTcpQueue<N> = self.get_shared_queue(&qd)?;
-        let coroutine_constructor = |yielder: Yielder| -> Result<TaskHandle, Fail> {
-            // Asynchronous accept code. Clone the self reference and move into the coroutine.
+        let coroutine_constructor = || -> Result<TaskHandle, Fail> {
+            let task_name: String = format!("inetstack::tcp::accept for qd={:?}", qd);
+            let yielder: Yielder = Yielder::new();
+            let yielder_handle = yielder.get_handle();
             let coroutine: Pin<Box<Operation>> = Box::pin(self.clone().accept_coroutine(qd, yielder));
-            // Insert async coroutine into the scheduler.
-            let task_name: String = format!("Catnap::accept for qd={:?}", qd);
-            self.runtime.insert_coroutine(&task_name, coroutine)
+            self.runtime
+                .insert_coroutine_with_tracking(&task_name, coroutine, yielder_handle, qd)
```

Comment:
This code snippet is used several times in this file. Does it make sense to create a priva function to handle this task?

Reply:
It is possible to do it for some cases, however the coroutine signature is different, so we may either need to generalize a lot or have specific functions that cater to them. So, for now I will just leave it the way it is.

The plan is to revisit this area to control places where Yielder/YielderHandles are created. So, we can address it separately.